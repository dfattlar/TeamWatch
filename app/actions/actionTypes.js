"use strict";

export const START_WATCH = "START_WATCH";
export const STOP_WATCH = "STOP_WATCH";
export const TICK = "TICK";
export const RESET_ALL = "RESET_ALL";
export const RESET_TIME = "RESET_TIME";
export const OPEN_MODAL = "OPEN_MODAL";
export const CLOSE_MODAL = "CLOSE_MODAL";
export const NEW_ATHLETE_INPUT = "NEW_ATHLETE_INPUT";
export const ADD_ATHLETE = "ADD_ATHLETE";
export const ADD_ATHLETE_ERROR = "ADD_ATHLETE_ERROR";
export const ADD_SPLIT = "ADD_SPLIT";
export const MODE_CHANGE = "MODE_CHANGE";
export const RESET_ATHLETE_LIST = "RESET_ATHLETE_LIST";
export const ADD_HISTORY = "ADD_HISTORY";
export const REMOVE_ATHLETE_FROM_WATCH = "REMOVE_ATHLETE_FROM_WATCH";
export const ADD_ATHLETE_TO_WATCH = "ADD_ATHLETE_TO_WATCH";
export const DELETE_ATHLETE = "DELETE_ATHLETE";
export const DELETE_HISTORY = "DELETE_HISTORY";
