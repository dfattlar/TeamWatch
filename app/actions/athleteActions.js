import * as types from './actionTypes';

export function addAthlete() {
    return {
        type: types.ADD_ATHLETE
    }
}

export function addAthleteToWatch(id, name) {
    return {
        type: types.ADD_ATHLETE_TO_WATCH,
        payload: { id,name }
    }
}

export function removeAthleteFromWatch(id) {
    return {
        type: types.REMOVE_ATHLETE_FROM_WATCH,
        payload: { id }
    }
}


export function addAthleteError() {
    return {
        type: types.ADD_ATHLETE_ERROR
    }
}

export function newAthleteInput(name) {
    return {
        type: types.NEW_ATHLETE_INPUT,
        newAthleteInput: name
    }
}
