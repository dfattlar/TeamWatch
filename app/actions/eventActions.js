"use strict";

import * as types from "./actionTypes";

export function deleteEvent(id) {
  return {
    type: types.DELETE_EVENT,
    id
  };
}
