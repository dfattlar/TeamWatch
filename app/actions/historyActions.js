'use strict';

import * as types from './actionTypes';

export function deleteHistory(id) {
    return {
        type: types.DELETE_HISTORY,
        id
    }
}
