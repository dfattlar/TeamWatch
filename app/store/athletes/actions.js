import * as types from './actionTypes';

export const addAthleteToWatch = (id, name) => {
  return dispatch => {
    dispatch(addingAthleteToWatch(id, name));
  };
};

export const removeAthleteFromWatch = id => {
  return dispatch => {
    dispatch(removingAthleteFromWatch(id));
  };
};

export const removeAllAthletesFromWatch = () => {
  return dispatch => {
    dispatch(removingAllAthletesFromWatch());
  };
};

export const createAthlete = (first, last) => {
  return dispatch => {
    dispatch(creatingAthlete(first, last));
  };
};

export const createAthleteError = error => {
  return dispatch => {
    dispatch(creatingAthleteError(error));
  };
};

const addingAthleteToWatch = (id, name) => ({
  type: types.ADD_ATHLETE_TO_WATCH,
  payload: {id, name},
});

const removingAthleteFromWatch = id => ({
  type: types.REMOVE_ATHLETE_FROM_WATCH,
  id,
});

const removingAllAthletesFromWatch = () => ({
  type: types.REMOVE_ALL_ATHLETES_FROM_WATCH,
});

const creatingAthlete = (first, last) => ({
  type: types.CREATE_ATHLETE,
  payload: {first, last},
});

const creatingAthleteError = error => ({
  type: types.CREATE_ATHLETE_ERROR,
  error,
});
