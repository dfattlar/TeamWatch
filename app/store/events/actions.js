import * as types from "./actionTypes";

export const addEvent = payload => {
  return dispatch => {
    dispatch(addingEvent(payload));
  };
};

const addingEvent = payload => ({
  type: types.ADD_EVENT,
  payload
});
