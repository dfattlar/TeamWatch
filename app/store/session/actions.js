import * as types from "./actionTypes";
import firebase from "react-native-firebase";

export const restoreSession = () => {
  return dispatch => {
    dispatch(sessionRestoring());

    let unsubscribe = firebase.auth().onAuthStateChanged(user => {
      if (user) {
        dispatch(sessionSuccess(user));
        unsubscribe();
      } else {
        dispatch(sessionLogout());
        unsubscribe();
      }
    });
  };
};

export const loginUser = (email, password) => {
  return dispatch => {
    dispatch(sessionLoading());

    firebase
      .auth()
      .signInWithEmailAndPassword(email, password)
      .catch(error => {
        dispatch(sessionError(error.message));
      });

    let unsubscribe = firebase.auth().onAuthStateChanged(user => {
      if (user) {
        dispatch(sessionSuccess(user));
        unsubscribe();
      }
    });
  };
};

export const signupUser = (email, password) => {
  return dispatch => {
    dispatch(sessionLoading());

    firebase
      .auth()
      .createUserWithEmailAndPassword(email, password)
      .catch(error => {
        dispatch(sessionError(error.message));
      });

    let unsubscribe = firebase.auth().onAuthStateChanged(user => {
      if (user) {
        dispatch(sessionSuccess(user));
        unsubscribe();
      }
    });
  };
};

export const logoutUser = () => {
  return dispatch => {
    dispatch(sessionLoading());

    firebase
      .auth()
      .signOut()
      .then(() => {
        dispatch(sessionLogout());
      })
      .catch(error => {
        dispatch(sessionError(error.message));
      });
  };
};

const sessionRestoring = () => ({
  type: types.SESSION_RESTORING
});

const sessionLoading = () => ({
  type: types.SESSION_LOADING
});

const sessionSuccess = user => ({
  type: types.SESSION_SUCCESS,
  user
});

const sessionError = error => ({
  type: types.SESSION_ERROR,
  error
});

const sessionLogout = () => ({
  type: types.SESSION_LOGOUT
});
