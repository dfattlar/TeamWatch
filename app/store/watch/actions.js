import * as types from './actionTypes';

export const startWatch = () => {
  return dispatch => {
    dispatch(startingWatch());
  };
};

export const stopWatch = () => {
  return dispatch => {
    dispatch(stoppingWatch());
  };
};

export const modeChange = timerMode => {
  return dispatch => {
    dispatch(changingMode(timerMode));
  };
};

export const resetAll = () => {
  return dispatch => {
    dispatch(resettingAll());
  };
};

export const resetTime = () => {
  return dispatch => {
    dispatch(resettingTime());
  };
};

export const addSplit = id => {
  return dispatch => {
    dispatch(addingSplit(id));
  };
};

const startingWatch = () => ({
  type: types.START_WATCH,
  startTime: Date.now(),
});

const stoppingWatch = () => ({
  type: types.STOP_WATCH,
  stopTime: Date.now(),
});

const changingMode = timerMode => ({
  type: types.MODE_CHANGE,
  timerMode,
});

const resettingAll = timerMode => ({
  type: types.RESET_ALL,
});

const resettingTime = timerMode => ({
  type: types.RESET_TIME,
});

const resettingAthletes = null; // TODO - need resetting athletes?

const addingSplit = id => ({
  type: types.ADD_SPLIT,
  id,
  splitTime: Date.now(),
});
