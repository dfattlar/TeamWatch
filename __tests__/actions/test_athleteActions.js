import * as actions from '../../app/actions/athleteActions';
import * as types from '../../app/actions/ActionTypes';

describe('Athlete Actions', () => {
  it('should create an action to input a new athlete', () => {
    const newAthleteInput = 'Athlete1';
    const expectedAction = {
      type: types.NEW_ATHLETE_INPUT,
      newAthleteInput,
    };

    expect(actions.newAthleteInput(newAthleteInput)).toMatchSnapshot();
  });

  it('should create an action to add an athlete', () => {
    const expectedAction = {
      type: types.ADD_ATHLETE,
    };
    expect(actions.addAthlete()).toEqual(expectedAction);
  });

  it('should create an action to add an athlete error', () => {
    const expectedAction = {
      type: types.ADD_ATHLETE_ERROR,
    };
    expect(actions.addAthleteError()).toEqual(expectedAction);
  });

  it('should create an action to add an athlete to the watch', () => {
    const id = 'id_123';
    const name = 'Athlete1';
    const expectedAction = {
      type: types.ADD_ATHLETE_TO_WATCH,
      payload: {
        id,
        name,
      },
    };
    expect(actions.addAthleteToWatch(id, name)).toEqual(expectedAction);
  });

  it('should create an action to remove an athlete from the watch', () => {
    const id = 'id_123';
    const expectedAction = {
      type: types.REMOVE_ATHLETE_FROM_WATCH,
      id,
    };
    expect(actions.removeAthleteFromWatch(id)).toEqual(expectedAction);
  });

  it('should create an action to delete an athlete', () => {
    const id = 'id_123';
    const expectedAction = {
      type: types.DELETE_ATHLETE,
      id,
    };
    expect(actions.deleteAthlete(id)).toEqual(expectedAction);
  });
});
