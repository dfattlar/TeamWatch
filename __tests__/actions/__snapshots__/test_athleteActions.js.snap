// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Athlete Actions should create an action to input a new athlete 1`] = `
Object {
  "newAthleteInput": "Athlete1",
  "type": "NEW_ATHLETE_INPUT",
}
`;
